session.go

```diff
@@ -151,14 +150,20 @@ func (s *session) recvLoop() {
 	stoppedOnExpectedEOF := false
 
 	defer func() {
+		log.Debugf("Closing lampshade TCP connection: %#v", s.pendingSession)
+		if s.pendingSessions != nil {
+			log.Debugf("Requesting new session: %#v", s.pendingSession)
+			s.pendingSessions <- s.pendingSession
```

Comment:
What if pendingSessions is full?

Reply:
`pendingSessions` is continually offloaded in `maintainTCPConnections`, so I don't think that can happen.